src/open-url.coffee

```diff
@@ -30,90 +30,92 @@ module.exports = (robot) ->
 				bookmarks = JSON.parse body
 
 	###*
-	* This creates a function to shim the http response
-	* @param {function} post successes to
-	* @param {function} post failures to
-	* @return {function} Takes (err, res, body) and passes them to (resolve, reject)
+	* Creates a function that converts a Hubot HTTP response into a Promise
+	* @param {function} resolve - a function for successful http requests
+	* @param {function} reject - a function for http errors
+	* @return {function} - a function that converts Hubot http responses
+	* (err, res, body) and renders them to a Promise function (resolve, reject)
 	###
 	createResolver = (resolve, reject) ->
 		(err, res, body) ->
 			if (not err?) and (res.statusCode is 200)
 				resolve(body)
 			else
-				reject(err ? new Error("StatusCode: #{res.statusCode}"))
+				reject(err ? new Error("StatusCode: #{res.statusCode}; Body: #{body}"))
 
 	###*
-	* This creates a function to report errors
-	* @param {Object} Hubot msg object
-	* @return {function} Takes (Error) and reports it
+	* Creates a function that communicates an error.
+	* @param {Object} context - A Hubot msg object
+	* @return {function} - a function that takes (error) and communicates it
 	###
 	createErrorReporter = (context) ->
 		(error) ->
 			robot.logger.error(error)
 			context.send('Something went wrong. Debug output logged')
 
 	###*
-	* Attempt to visit the url referenced by key
-	* @param {string} url to visit
-	* @return {Promise} Handle output from the asynchronous
+	* Attempt to finger the url referenced by key using GET
+	* @param {string} url - address of the web site to finger
+	* @return {Promise} - A promise for the response for the given url
 	###
-	open = (url) ->
+	finger = (url) ->
```

Comment:
`finger` is a strange word to use. Weird connotations, and historically it's a [separate protocol](https://en.wikipedia.org/wiki/Finger_protocol) that does something else (gets user details).

`check`/`test`? I'd suggest `ping`, but of course that's plausibly confusing too.